server/controllers/auth.js

```diff
@@ -0,0 +1,128 @@
+                    console.log(
+                        `req.session.passport: ${JSON.stringify(
+                            req.session.passport
+                        )}`
+                    );
+                    console.log(`req.user: ${JSON.stringify(req.user)}`);
+                    res.redirect(returnTo || '/user'); // client side
+                });
+            })(req, res, next);
+        } catch (err) {
+            console.log(err);
+            res.status(INTERNAL_SERVER_ERROR).send({
+                status: ERROR,
+                message: HttpStatus.getStatusText(INTERNAL_SERVER_ERROR),
+            });
+        }
+    },
+    login: async (req, res) => {
+        try {
+            res.redirect('/');
+        } catch (err) {
+            res.status(INTERNAL_SERVER_ERROR).send({
+                status: ERROR,
+                message: HttpStatus.getStatusText(INTERNAL_SERVER_ERROR),
+            });
+        }
+    },
+    logout: async (req, res) => {
+        try {
+            await req.logout();
+            let returnTo = `${req.protocol}://${req.hostname}`;
+            const port = req.connection.localPort;
+            if (port !== undefined && port !== 80 && port !== 443) {
+                returnTo += `:${port}`;
+            }
+            const logoutURL = new URL(
+                util.format('https://%s/logout', process.env.AUTH0_DOMAIN)
+            );
+            const searchString = querystring.stringify({
+                client_id: process.env.AUTH0_CLIENT_ID,
+                returnTo,
+            });
+            logoutURL.search = searchString;
+
+            res.redirect(logoutURL);
+            // res.redirect('/home');
+        } catch (err) {
+            res.status(INTERNAL_SERVER_ERROR).send({
+                status: ERROR,
+                message: HttpStatus.getStatusText(INTERNAL_SERVER_ERROR),
+            });
+        }
+    },
+    test: async (req, res) => {
+        try {
+            res.status(200).json({ msg: 'user works' });
+        } catch (err) {
+            res.status(INTERNAL_SERVER_ERROR).send({
+                status: ERROR,
+                message: HttpStatus.getStatusText(INTERNAL_SERVER_ERROR),
+            });
+        }
+    },
+    users: async (req, res) => {
+        try {
+            console.log(req.session.id);
+            console.log(req.session.cookie);
+            const isAuthenticated = await req.isAuthenticated();
+            console.log(`User authenticated? ${isAuthenticated}`);
+            if (isAuthenticated) {
+                const user = await req.user;
+
+                console.log(user);
+                res.status(OK).send({
+                    status: SUCCESS,
+                    data: { user },
+                });
+            } else {
+                res.status(UNAUTHORIZED).send({
+                    status: FAIL,
+                    data: { user: 'User Not Authorized' },
```

Comment:
send `message: HttpStatus.getStatusText(UNAUTHORIZED)` instead of `data`

Reply:
I am not sure but with the Jsend API convention when the status is fail or success you send data but when the status is error you send a message